controllers/grid/SubmissionListGridCellProvider.inc.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * @file controllers/grid/SubmissionListGridCellProvider.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionListGridCellProvider
+ * @ingroup controllers_grid_sword
+ *
+ * @brief Class for a cell provider to display information about a submission
+ */
+import('lib.pkp.classes.controllers.grid.GridCellProvider');
+import('lib.pkp.classes.linkAction.request.RedirectAction');
+
+class SubmissionListGridCellProvider extends GridCellProvider {
```

Comment:
I suspect some of the other import/export plugins that allow selection of submissions from a list already have an implementation that could be used here too.

Reply:
I remember looking around the codebase but was unable to find any.

Reply:
Have you looked at the article listings in the import/export tools?

Reply:
Ah, yes, I'd forgotten that those were converted. @NateWr, are the submission lists in import/export easy to reuse for a new plugin? If yes, got a couple of quick pointers into the code for how to do so?

Reply:
Yes! Is this just a select a submission list? If so, there's a `SelectSubmissionsListPanel` component that can be used.

You can check this out and get some documentation on it in the UI Library:

```
cd lib/ui-library
npm install
npm run dev
// browse to SelectListPanel > Implementations > SelectSubmissionsListPanel
```

In PHP-land, there's a `SubmissionsListHandler`, which you can invoke and pass to the template like this:

```
$selectSubmissionsListHandler = new SelectSubmissionsListHandler($args);
$selectSubmissionsConfig = $selectSubmissionsListHandler->getConfig();
$templateMgr->assign('selectSubmissionsListData', json_encode($selectSubmissionsConfig);
```

If you need to adjust the parameters with which it fetches submissions (for example, limiting to published submissions, just one section, etc), you can pass a `getParams` arg:

```
$selectSubmissionsListHandler = new SelectSubmissionsListHandler(array('getParams' => array(...));
```

It will accept any args supported by the API, so you can check the documentation there. Finally, to actually spin  up the Vue.js component in the template, you can do the following:

```
{assign var="uuid" value=""|uniqid|escape}
<div id="select-submissions-list-handler-{$uuid}">
	<script type="text/javascript">
		pkp.registry.init('export-submissions-list-handler-{$uuid}', 'SelectSubmissionsListPanel', {$selectSubmissionsListData});
	</script>
</div>
```

And you _should_ have your list panel up on the form. :+1:

Reply:
Thanks @NateWr. I have successfully replaced my custom grid with your select submission panel.